pkg/upgrade/upgrade_test.go

```diff
@@ -1,17 +1,24 @@
-package upgrade_test
+package upgrade
 
 import (
+	"fmt"
 	"testing"
 
-	"github.com/onsi/ginkgo"
-	"github.com/onsi/ginkgo/extensions/table"
-	"github.com/onsi/gomega"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift-psap/special-resource-operator/pkg/cache"
 	"github.com/openshift-psap/special-resource-operator/pkg/registry"
-	"github.com/openshift-psap/special-resource-operator/pkg/upgrade"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
-var _ = ginkgo.Describe("pkg/upgrade", func() {
-	ginkgo.Context("UpdateInfo", func() {
+var _ = Describe("ClusterInfo", func() {
+
+	clusterInfo := NewClusterInfo().(*clusterInfo)
+
+	Context("updateInfo", func() {
```

Comment:
I don't think we want to test private functions. Could we refactor the tests to integrate those cases into the public method's?

Reply:
That's a good idea.
The thing is that `driverToolkitVersion()` depends on that very heavily on package `registry` (which e.g. creates pull secret on disk, uses `github.com/google/go-containerregistry/pkg/v1` and `github.com/google/go-containerregistry/pkg/crane`).

What do you think about refactoring `registry` first and using its mock here?

Reply:
That makes sense. Do you want to wait until then before we merge this?

Reply:
Yes. I will refactor the `registry` without UTs and after its merge use it here.